server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -224,7 +225,17 @@ public ClusterState execute(ClusterState currentState) {
                 }
                 // Store newSnapshot here to be processed in clusterStateProcessed
                 List<String> indices = Arrays.asList(indexNameExpressionResolver.concreteIndexNames(currentState,
-                    request.indicesOptions(), request.indices()));
+                    request.indicesOptions(), true, request.indices()));
+
+                Map<String, DataStream> allDataStreams = currentState.metadata().dataStreams();
+                List<DataStream> dataStreams;
+                if(request.includeGlobalState()){
+                    dataStreams = new ArrayList<>(allDataStreams.values());
+                } else {
+                    dataStreams = indexNameExpressionResolver.dataStreamNames(currentState, request.indicesOptions(),
```

Comment:
You have to make sure not to allow any specific data stream snapshotting as long as there's master nodes older than `DATA_STREAMS_IN_SNAPSHOT` in the cluster. Otherwise, if you have a master failover to an old version master the information about the data stream snapshots is just lost because it won't deserialize that part.

Reply:
Thanks for suggestion, is there a way to check that whole cluster is already on correct version?

Reply:
Jup, you can use: 

```java
clusterState.nodes().getMinNodeVersion()
```

see e.g. 
https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java#L231

Reply:
I think the `version` variable indicates the lowest version of nodes in the entire cluster.
(that variable is based on `clusterService.state().nodes().getMinNodeVersion()`)

Reply:
stale read from side...

Reply:
Added required check